docs/guides/Guide-Headers.md

```diff
@@ -79,4 +79,61 @@ Now, the header can interact with the screen route/state:
 header-interaction
 ```
 
+### Header interaction with screen component
+
+Sometimes it is necesarry for header to access properties of the screen component such as functions or state.
+
+Let's say we want to create an 'edit contact info' screen with a save button in the header. We want the save button to be replaced by an `ActivityIndicator` while saving.
+
+```js
+class EditInfoScreen extends React.Component {
+  static navigationOptions = ({ navigation }) => {
+    const { params = {} } = navigation.state;
+    let headerRight = (
```

Comment:
Why not use a ternary with `const` instead of reassignment with let?

Reply:
I think using ternary in the return object gets confusing, but we could do:

```js
static navigationOptions = ({ navigation }) => {
  const { params = {} } = navigation.state;
  const headerRight = params.isSaving ? <ActivityIndicator /> : (
    <Button
      title="Save"
      onPress={params.handleSave ? params.handleSave : () => null}
    />
  );
  return { headerRight };
};
```

Would you prefer this?

Reply:
I agree with @Jacse in this scenario: since it's an example, it's easier for a reader to follow what's happening if there is a big "if statement" clearly telling you what's going to be further explained - in the phrase after the code block.